packages/identity-secp256k1/src/pem.ts

```diff
@@ -0,0 +1,42 @@
+import { concat, uint8ToBuf } from '@dfinity/agent';
+
+const HEADER = `-----BEGIN`;
+const FOOTER = `-----END`;
+const ECHEADER = `-----BEGIN EC PRIVATE KEY`;
+
+/**
+ * Parse a PEM-encoded key into an ArrayBuffer
+ * @param pem - the PEM-encoded key
+ * @returns secret key as an ArrayBuffer
+ */
+export function pemToSecretKey(pem: string): ArrayBuffer {
+  const lines = pem.trim().split('\n');
+  const header = lines[0].trim();
+  const footer = lines[lines.length - 1].trim();
+  if (lines.length < 3) {
+    throw new Error('Invalid PEM format');
+  }
+  if (!header.startsWith(HEADER)) {
+    throw new Error('Invalid PEM header');
+  }
+  if (!footer.startsWith(FOOTER)) {
+    throw new Error('Invalid PEM footer');
+  }
+  const base64Data = lines.slice(1, -1).join('').replace(/\r?\n/g, '');
+  const rawKey = Uint8Array.from(atob(base64Data), c => c.charCodeAt(0));
+
+  if (pem.startsWith(ECHEADER)) {
+    if (rawKey.length !== 118) {
+      throw new Error(`Invalid key length ${rawKey.length}. Expected 118 bytes.`);
+    } else {
+      return rawKey.slice(7, 39);
```

Comment:
I guess this is just extracting the raw 32 byte binary key at the appropriate offset? Probably worth a comment.